pkgs/by-name/eb/ebpf-usb/package.nix

```
@@ -0,0 +1,50 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, bcc
+, makeWrapper
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ebpf-usb";
+  version = "unstable-2022-04-03";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "francisrstokes";
+    repo = "ebpf-usb";
+    rev = "3ab6f0d8c6ece51bbb5cc5e05daa4008eccd70e8";
+    sha256 = "n3ttFej9sroTqAOgyAejwKT+aMt/z7HlVPV6CVGPNUQ=";
+  };
+
+  nativeBuildInputs = [ bcc makeWrapper ];
+
+  propagatedBuildInputs = with python3Packages; [
+    hexdump
+  ];
+
+  # No setup.py
+  dontBuild = true;
+
+
+  installPhase = let
+    pythonVersion = lib.versions.majorMinor python3Packages.python.version;
+  in
+  ''
+    install -Dm755 ebpf-usb.py $out/bin/ebpf-usb
+    BCC_SITE_PACKAGES=${bcc}/lib/python${pythonVersion}/site-packages/bcc-${bcc.version}-py${pythonVersion}.egg
+    wrapProgram $out/bin/ebpf-usb \
+      --prefix PYTHONPATH : $PYTHONPATH:$BCC_SITE_PACKAGES
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 ebpf-usb.py $out/bin/ebpf-usb
    substituteInPlace $out/bin/ebpf-usb --replace \
      '#!/usr/bin/env -S python3 -u' '#!/usr/bin/env python3'
    runHook postInstall
  '';
```

wrapPythonPrograms will get called in the postFixup but the arguments are preventing that, so remove them and use env variables to set unbuffered input.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md